packages/functional-tests/pages/signupReact.ts

```diff
@@ -1,118 +1,134 @@
+import { expect } from '@playwright/test';
 import { BaseLayout } from './layout';
 import { getReactFeatureFlagUrl } from '../lib/react-flag';
 import { EmailHeader, EmailType } from '../lib/email';
 
+export type EmailFormData = {
+  email: string;
+  submit: boolean;
+};
+
+export type SignupFormData = {
+  password: string;
+  age: string;
+  submit: boolean;
+};
+
+export type CodeFormData = {
+  email: string;
+  submit: boolean;
+};
+
 export class SignupReactPage extends BaseLayout {
   readonly path = 'signup';
 
-  goto(route = '/', params = new URLSearchParams()) {
-    params.set('forceExperiment', 'generalizedReactApp');
-    params.set('forceExperimentGroup', 'react');
-    return this.page.goto(
-      getReactFeatureFlagUrl(this.target, route, params.toString())
-    );
+  get emailFormHeading() {
+    return this.page.getByRole('heading', {
+      name: /^Enter your email|^Continue to your Mozilla account/,
+    });
   }
 
-  async setEmail(value: string) {
-    const emailInput = await this.getEmail();
-    return emailInput.fill(value);
+  get emailTextbox() {
+    return this.page.getByRole('textbox', { name: 'Enter your email' });
   }
 
-  getEmail() {
-    return this.page.getByRole('textbox', { name: 'email' });
+  get submitButton() {
+    return this.page.getByRole('button', { name: 'Sign up or sign in' });
   }
 
-  async setPassword(value: string) {
-    const newPasswordInput = this.getPassword();
-    return newPasswordInput.fill(value);
+  get signupFormHeading() {
+    return this.page.getByRole('heading', { name: 'Set your password' });
   }
 
-  getPassword() {
-    return this.page.getByRole('textbox', {
-      name: 'Password',
-      exact: true,
-    });
+  get passwordTextbox() {
+    return this.page.getByTestId('new-password-input-field');
```

Comment:
Man I really dislike all these test IDs 🙈 (nothing to change in your PR, just lamenting about us using these heavily in a few spots)

Reply:
Yes. I agree this is not ideal. It will also be a problem if we want to use the accessibility analysis Playwright supports in future if I'm reading things correctly.